tailer/__init__.py

```diff
@@ -160,7 +161,17 @@ def follow(self, delay=1.0):
         """
         trailing = True
 
+        initial_inode = os.stat(self.file.name).st_ino
+        file_path = self.file.name
+
         while 1:
+            try:
+                if os.stat(self.file.name).st_ino != initial_inode:
+                    self.file = open(file_path)
+                    initial_inode = os.stat(self.file.name).st_ino
+            except FileNotFoundError:
+                continue
```

Comment:
This will get stuck in a tight infinite loop if the file does not exist. We need a non-blocking wait. The easiest way to do this is to introduce a `time.sleep(delay)` if the file does not exist. 

See comments on https://github.com/six8/pytailer/pull/14/files#r1167325214

We want to mimic what GNU tail is doing. 

With GNU tail, the `-f` option will follow the file by inode. So if the file is renamed, it will continue to follow the file at the new name. We want to continue that behavior. 

GNU tail adds a `-F` option, which is an alias for `--follow=name --retry`

*Note:* In `--follow=name`, `name` is a constant value. The options are `name` or `descriptor` with `descriptor` being the default.

See the GNU tail docs for more info https://man7.org/linux/man-pages/man1/tail.1.html

The following tests needed to be added. Start with the tests, then the implementation will be more obvious:

1. If using `-f` and the file is renamed, it continues to follow the file at the new name and ignores a file at the old name.
2. If using `-F` (or `--follow=name --retry`) and the file is renamed, ignore the new name file and follow the old name file.

This means adding a new `--retry` and `-F` option and making it so `--follow` has 3 states:

1. `--follow`
2. `--follow=name`
3. `--follow=descriptor`

You can see how GNU tail is implemented at https://github.com/coreutils/coreutils/blob/master/src/tail.c. It uses `inotify`, which we can't use because `inotify` is not part of the Python standard library, and the intention is to keep `tailer` pure-python and only depend on the standard library.